src/runtime/nitro/plugins/cspNonce.ts

```diff
@@ -0,0 +1,41 @@
+import type { NitroAppPlugin } from 'nitropack'
+import type { H3Event } from 'h3'
+import type {
+  ModuleOptions
+} from '../../../types'
+import { useRuntimeConfig } from '#imports'
+
+interface NuxtRenderHTMLContext {
+  island?: boolean
+  htmlAttrs: string[]
+  head: string[]
+  bodyAttrs: string[]
+  bodyPrepend: string[]
+  body: string[]
+  bodyAppend: string[]
+}
+
+export default <NitroAppPlugin> function (nitro) {
+  nitro.hooks.hook('render:html', (html: NuxtRenderHTMLContext, { event }: { event: H3Event }) => {
+    const nonce = parseNonce(`${event.node.res.getHeader('Content-Security-Policy')}`)
+
+    if (!nonce) { return }
+
+    // Add nonce attribute to all link tags
+    html.head = html.head.map(link => link.replaceAll(/<link/g, `<link nonce="${nonce}"`))
+    html.bodyAppend = html.bodyAppend.map(link => link.replaceAll(/<link/g, `<link nonce="${nonce}"`))
+
+    // Add nonce attribute to all script tags
+    html.head = html.head.map(script => script.replaceAll(/<script/g, `<script nonce="${nonce}"`))
```

Comment:
What about something like this:

```html
<script>var myVar = '<script>'</script>
```

I mean if `<script` text will be contained inside script. If I do not mistake, evety value in html.head (and others) array is single script or link element. May be regex should be `/^<link/g` (with `^` symbol).

Reply:
I was unaware of this property of `html.head`!
If it's indeed guaranteed that every value in `html.head` is a single element then your suggestion makes a lot of sense.

A more simple approach would be to just use `replace` instead of `replaceAll` in that case. Probably a tiny bit quicker than using a regex.

Reply:
One of examples in this discussion comment: https://github.com/nuxt/nuxt/issues/18376#issuecomment-1438303970

Can it help you better?

Reply:
I'm not sure I see the problem with the current implementation.

Your example of :

```
<script>var myVar = '<script>'</script>
```

would not be allowed when using nonce, unless you also add `strict-dynamic`  to CSP.
Otherwise, it will need a `nonce` attribute in any case, as it otherwise won't be adhering to the CSP policy.

The current implementation will add the nonce to that inside script tag as well, which would work with or without `strict-dynamic`.

I personally would wait with adding extra parsing logic (and thus increasing code complexity) until it shows that people are having problems with the chosen approach.

Still, you are of course free to add or change this PR if you feel that's necessary in this case.

Reply:
In my example I mean that your solution also will replace string inside content of script element. But someone might to use `<script>` text for... I don not know... in text for examples in documentation on page of some site. It will be just text, but `/<script/g` regex replace and it too.

Reply:
Yeah I understand what you mean, but is that really a real-world scenario? We're talking about a nuxt module for a nuxt app here: someone would have to go out of their way to add a script like that in the head tag instead of using a component.

Furthermore, they then start using this nuxt-security module, and opt-in to the nonce behaviour: a conscious decision that you only do if you care about enforcing a strict csp policy. In any case they should be thinking about these scripts and why they are loading them in this non-nuxtonian-way.

I just can't fanthom that being a real world scenario.

Still, I can add a test for your specific situation and update the RegExp accordingly. IMO the solution is to check that there are no quotes (`'` or `"`) around the script tag.

Reply:
See https://github.com/Baroshem/nuxt-security/pull/171/commits/828387ef16b105141fbb2a7a8b4e38a72718258e

Reply:
Thank you very much 😊

My worried soul will be calm)))